training/model_management/components/import_model/spec.yaml

```diff
@@ -0,0 +1,135 @@
+$schema: https://azuremlschemas.azureedge.net/latest/pipelineComponent.schema.json
+type: pipeline
+
+name: import_model
+display_name: Import model
+description: Import a model into a workspace or a registry
+version: 0.0.1
+
+# Pipeline inputs
+inputs:
+  # pipeline specific compute
+  compute:
+    type: string
+    description: Compute to run pipeline job
+
+  ## Inputs for download model
+  model_source:
+    type: string
+    description: Storage containers from where model will be sourced from.
+    default: Huggingface
+    enum:
+      - AzureBlob
+      - GIT
+      - Huggingface
+
+  model_id:
+    type: string
+    description: A valid model id for the model source selected. For example you can specify `bert-base-uncased` for importing HuggingFace bert base uncased model. Please specify the complete URL if **GIT** or **AzureBlob** is selected in `model_source`
+
+  ## Inputs for the MlFLow conversion
+  license_file_path:
+    type: uri_file
+    description: Path to the license file
+    optional: true
+
+  ## Inputs for model local validation
+  test_data_path:
+    type: uri_file
+    optional: true
+    description: Test dataset for model inferencing
+
+  test_data_column_rename_map:
+    type: string
+    optional: true
+    description: |
+      Provide mapping of dataset column names that should be renamed before inferencing.
+      eg: col1:ren1; col2:ren2; col3:ren3
```

Comment:
local validation & inference do not need to be part of the import model. Let us include them in later phases or release separate components.

Reply:
We will not be able to figure out the right SKU needed for the model and if incompatible, user experience will suffer due to the pipeline itself failing.

Reply:
@novaturient95 Can we do an if/else based on whether the test data is passed?

Reply:
Yes, we can do that. `test_file` is an optional input. If its not present we would not load the model and let it go to register model

Reply:
Let us discuss this as well internally and figure out. 

For now, let us not include local validation & data input in the component.